include/pipettes/core/tasks/motion_controller_task.hpp

```diff
@@ -0,0 +1,156 @@
+#include "pipettes/core/tasks/messages.hpp"
+
+namespace pipettes {
+
+namespace tasks {
+
+namespace motion_controller_task {
+
+using TaskMessage = pipettes::task_messages::motor_control_task_messages::
+    MotionControlTaskMessage;
+
+/**
+ * The message queue message handler.
+ */
+template <lms::MotorMechanicalConfig MEConfig,
+          message_writer_task::TaskClient CanClient>
+class MotionControllerMessageHandler {
+  public:
+    using MotorControllerType =
+        pipette_motion_controller::PipetteMotionController<MEConfig>;
+    MotionControllerMessageHandler(MotorControllerType& controller,
+                                   CanClient& can_client)
+        : controller{controller}, can_client{can_client} {}
+    MotionControllerMessageHandler(const MotionControllerMessageHandler& c) =
+        delete;
+    MotionControllerMessageHandler(const MotionControllerMessageHandler&& c) =
+        delete;
+    auto operator=(const MotionControllerMessageHandler& c) = delete;
+    auto operator=(const MotionControllerMessageHandler&& c) = delete;
+    ~MotionControllerMessageHandler() = default;
+
+    void handle_message(const TaskMessage& message) {
+        std::visit([this](auto m) { this->handle(m); }, message);
+    }
+
+  private:
+    void handle(std::monostate m) { static_cast<void>(m); }
+
+    void handle(const can_messages::StopRequest&) {
+        LOG("Received stop request");
+        controller.stop();
+    }
+
+    void handle(const can_messages::EnableMotorRequest&) {
+        LOG("Received enable motor request");
+        // TODO only toggle the enable pin once since all motors share
+        // a single enable pin line.
+        controller.enable_motor();
+    }
+
+    void handle(const can_messages::DisableMotorRequest&) {
+        LOG("Received disable motor request");
+        // TODO only toggle the enable pin once since all motors share
+        // a single enable pin line.
+        controller.disable_motor();
+    }
+
+    void handle(const can_messages::GetMotionConstraintsRequest&) {
+        auto constraints = controller.get_motion_constraints();
+        can_messages::GetMotionConstraintsResponse response_msg{
+            .min_velocity = constraints.min_velocity,
+            .max_velocity = constraints.max_velocity,
+            .min_acceleration = constraints.min_acceleration,
+            .max_acceleration = constraints.max_acceleration,
+        };
+        LOG("Received get motion constraints request");
+        can_client.send_can_message(can_ids::NodeId::host, response_msg);
+    }
+
+    void handle(const can_messages::SetMotionConstraints& m) {
+        LOG("Received set motion constraints: minvel=%d, maxvel=%d, minacc=%d, "
+            "maxacc=%d",
+            m.min_velocity, m.max_velocity, m.min_acceleration,
+            m.max_acceleration);
+        controller.set_motion_constraints(m);
+    }
+
+    void handle(const can_messages::TipActionRequest& m) {
```

Comment:
So the gear motors can't home or do arbitrary moves, is that the idea? How do they home?

Reply:
A drop tip will continue moving the motors until the limit switches are triggered.